### PR TITLE
Add client certificate authentication to Azure cloud provider

### PR DESCRIPTION
**Pull request by ccojocar**
This adds support for client cert authentication in Azure cloud provider. The certificate can be provided in PKCS #12 format with password protection. Not that this authentication will be active only when no client secret is configured.

cc @brendandburns @colemickens

**Release note**:
```release-note
Add client certificate authentication to Azure cloud provider
```

### CODE REVIEW

**Comment by thockin** [comment]
@k8s-bot ok to test

**Comment by roberthbailey** [comment]
To clarify, this is for client cert auth to the Azure API, not from a k8s client to the k8s apiserver, correct?

**Comment by cosmincojocar** [comment]
This is correct. The client cert auth is for Azure API.

File: pkg/cloudprovider/providers/azure/azure_test.go [hunk 1]
```diff
@@ -585,6 +585,8 @@ func TestNewCloudFromJSON(t *testing.T) {
 		"subscriptionId": "--subscription-id--",
 		"aadClientId": "--aad-client-id--",
 		"aadClientSecret": "--aad-client-secret--",
+                "aadClientCertPath": "--aad-client-cert-path--",
```
Comment (brendandburns) [inline pkg/cloudprovider/providers/azure/azure_test.go]: nit: spacing is weird here (one too many indents) (and line below)

**Comment by brendandburns** [comment]
This looks good to me.

You need to run `./hack/update-bazel.sh` and I have one nit about formatting...

Then I can lgtm.

Thanks!

File: pkg/cloudprovider/providers/azure/azure_test.go (hunk 1 above)
Comment (cosmincojocar) [inline pkg/cloudprovider/providers/azure/azure_test.go]: fixed

**Comment by cosmincojocar** [comment]
@brendanburns looks like the `lgtm` label disappeared after rebase. Please could you re-lable it?

**Comment by colemickens** [comment]
@cosmincojocar You check for `AADClientCertPassword` but you don't actually use it. It looks like you use the ClientSecret to decode the cert. Is this intentional? If not, I can fix it in a PR I'm authoring now.

**Comment by cosmincojocar** [comment]
@colemickens This is a bug. Thanks for catching it, it'll send a fix right now.

**Pull request merged**